### PR TITLE
Merge in Odia script

### LINKED ISSUE #515: Add Odia script support
Hello @madig, @belluzj, and @chrissimpkins 

Attaching the latest Google Sans Odia package here. The QA toml files will be available later today.

Please let me know if I need to fix anything.

Many thanks!

[2022-07-22_GoogleSansOdia_V1.000_BETA01.zip](https://github.com/googlefonts/googlesans/files/9167467/2022-07-22_GoogleSansOdia_V1.000_BETA01.zip)

### PR DESCRIPTION
**Pull request by Hoolean**
Initial import of Odia, pending some glyph names and testing strings.

Closes #515.

### CODE REVIEW

**Comment by chrissimpkins** [comment]
@Hoolean At the moment, our static instance QA checks only run when we push to a `build-*` prefixed branch.  Can we execute the static instance tests when you have a chance? Ty!

Note: this may be changing in #507.  @madig?

**Comment by Hoolean** [comment]
Hello Chris! I have now run the static checks (manually) [here](https://github.com/googlefonts/googlesans/runs/7581211135). I will also speak with Nikolaus about how best to have these run automatically with the upcoming GitHub actions.

The checks appear to pass all clear, with the exception of [GDEF warnings about marks](https://github.com/googlefonts/googlesans/runs/7581211135#step:7:216). Having compared, these warnings are identical to those from the variable QA checks that [Kalapi believed](https://github.com/googlefonts/googlesans/issues/515#issuecomment-1195560917) were false-positives.

**Comment by Hoolean** [comment]
In the spirit of expanding our regression tests (for #516 and similar), Nikolaus and I have been looking at extracting the test strings in the manual review document that Kalapi provided, and [housing them in further TOML shaping inputs](https://github.com/googlefonts/googlesans/pull/518/commits/280b3d160d2cf645fa8889673cdb41eb8595739e) for the automated fontbakery checks.

This is a cheap way of getting better coverage of our imported scripts, and could be applied in retrospect where other sample documents have been provided in the past. The cost comes from having [large and unwieldy JSON snapshots](https://github.com/googlefonts/googlesans/pull/518/commits/52e8d70534f6166738b6b6d55ac03d76028ba3d5), which lose readability and may contain redundancy, reaching a few megabytes in size.

@chrissimpkins Do you feel this is worth including for Odia? In addition, should we explore this for the other scripts that included further proof text, or would we be better focusing on more targeted test strings, or fuzz-based coverage? : )

@kalapi To help us trim these down a little, if you would be happy to comment: are the test strings in the TOML files you provided also included in the HTML QA page, or are they important to include separately (e.g. because they test certain glyphs or features that do not come up in the QA page otherwise)?

---

On a separate note, we have rebased this branch to include several changes from main, most notably the pruning of stale glyphs. Since doing so, we have re-run the static and variable tests, which passed with the same warnings as before.

**Comment by kalapi** [comment]
Hi @Hoolean,

The HTML-testers are systematic tests. On some occasions they may contain combinations that appear broken, but this is intentional, and intended to test the limits of shaping. The TOML files are a subset of the HTML-test strings, and have been chosen on the basis that if one or more fails to shape correctly the shaping is failing. 

I hope this make sense?

**Comment by chrissimpkins** [comment]
> @chrissimpkins Do you feel this is worth including for Odia?

IMO we should explore these ideas in the compiler performance improvement test suite design doc before we roll out a new approach.  I think that what you are getting at is the threshold beyond which we feel safe that we've appropriately covered some arbitrary level of possible layout issues.  This addition might get us closer, but is it close enough?  I'd like to understand that issue in the context of the compiler regression test suite discussion.

> fuzz-based coverage

That is a very interesting idea!  I think that Simon had a similar thought at one point.  Let's discuss.

I don't know if it applies in this case but we'll also need to make sure that we know the license status for any copyright material if we happen to move towards the addition of new types of text runs to support broadened regression testing.

**Comment by Hoolean** [comment]
Understood Chris; let's have a deeper think! I have included this in our next agenda, or we could discuss earlier too, if you like.

I have moved the changes from here into their own branch and PR (#520), so as not to unduly block the merge.

**Comment by chrissimpkins** [comment]
I spoke with Kalapi today.  He is prepared to re-review as soon as we hear if there are any other comments from Jonny.

**Comment by chrissimpkins** [comment]
We ready for review and merge here?

**Comment by chrissimpkins** [comment]
Will you please trigger the static CI checks on this branch as of https://github.com/googlefonts/googlesans/pull/518/commits/698985d0032f55b6ecf89445db217806c27489a2?

**Comment by Hoolean** [comment]
Hi Chris! I have pushed to a build branch to trigger the static CI checks, which are now [in progress](https://github.com/googlefonts/googlesans/actions/runs/2935074010
).

I will pass the TTFs that result to Kalapi on Monday, so we can check that `/aaMatra-oriya` is behaving correctly after we [patched its missing category](https://github.com/googlefonts/googlesans/pull/518/commits/698985d0032f55b6ecf89445db217806c27489a2); after this, we should be ready for review and merge. :+1:

The only other known caveats are the InDesign issue that we discussed, and a minor shaping approach difference I spotted in Crowbar (which caused no difference in the shaping end result, but which I might run by the Font Engineering and Tools group to reassure myself of its meaning).

**Comment by belluzj** [comment]
@chrissimpkins @MariannaPaszkowska I have rebased just now this branch (import-odia) and the build- branch (build-import-odia) onto main. Both branches point to the same commit.

**Comment by chrissimpkins** [comment]
New file sizes following the Odia merge:


```
build/GoogleSans/variable/GoogleSans[GRAD,opsz,wght].ttf

             Summary Statistics              
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃                    Statistic ┃   Size (B) ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━┩
│                        Total │ 3433088.00 │
│ Avg per encoded glyph (2877) │    1193.29 │
│   Avg per total glyph (5979) │     573.81 │
└──────────────────────────────┴────────────┘

build/GoogleSans/variable/GoogleSans-Italic[GRAD,opsz,wght].ttf

             Summary Statistics              
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃                    Statistic ┃   Size (B) ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━┩
│                        Total │ 3686020.00 │
│ Avg per encoded glyph (2877) │    1281.20 │
│   Avg per total glyph (5979) │     616.49 │
└──────────────────────────────┴────────────┘
```

File: docs/MAINTAINER.md [hunk 1]
```diff
@@ -256,6 +256,24 @@ $ ./update_all_shaping.sh
 
 The FontBakery QA tests will pick up the file automatically.
 
+### Updating glyphset definitions
```
Comment (chrissimpkins) [inline docs/MAINTAINER.md:259]: Thank you very much!

**Review by chrissimpkins — APPROVED** [review]
LGTM.  OK to merge from my standpoint @MariannaPaszkowska.

**Comment by chrissimpkins** [comment]
We'll address https://github.com/googlefonts/googlesans/issues/530 in a separate PR @kalapi

**Comment by chrissimpkins** [comment]
Thanks @kalapi @Hoolean @MariannaPaszkowska !

**Pull request merged**